### PR TITLE
shell_sort.kt

### PR DESCRIPTION
**Pull request by Awani-Shukla**
**Fixes issue:**

**Changes:**this is in kotlin.

### CODE REVIEW

File: code/sorting/src/shell_sort/shell_sort.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            if (this[index] > this[index + sublistCount]) {
+                swap(index, index + sublistCount)
+            }
+
+            if (sublistCount == 1 && index > 0) {
+                while (this[index - 1] > this[index] && index - 1 > 0) {
+                    swap(index - 1, index)
+                    index -= 1
+                }
+                index++
```
Comment (arnavb) [inline code/sorting/src/shell_sort/shell_sort.kt]: Preincrement.

File: code/sorting/src/shell_sort/shell_sort.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    this[index1] = this[index2]
+    this[index2] = tmp
+}
+
+fun MutableList<Int>.shellSort(): MutableList<Int> {
+
+    var sublistCount = count() / 2
+
+    while (sublistCount > 0) {
+
```
Comment (abdouskamel) [inline code/sorting/src/shell_sort/shell_sort.kt]: Useless empty line here.

File: code/sorting/src/shell_sort/shell_sort.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        var index = 0
+        outer@while (index >= 0 && index < count()) {
+
```
Comment (abdouskamel) [inline code/sorting/src/shell_sort/shell_sort.kt]: Same here.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**